traitsui/qt4/color_editor.py

```diff
@@ -289,7 +289,10 @@ def color_editor_for(editor, parent):
 
             i += 1
 
-    mapper.mapped[str].connect(editor.update_object_from_swatch)
+    if is_pyside:
+        mapper.mappedString.connect(editor.update_object_from_swatch)
```

Comment:
We don't care about PySide 1: it was Python 2 or early Python 3 only, and is not supported on any current Python we support.

I chose this split because it was the easiest to get the required behaviour - yes, some versions of different backends could have been on various sides of the conditional and still worked - and we may need to revisit for PyQt6 support as well.